MIDI/ambrosebs_MIDI Drum Trainer.jsfx

```diff
@@ -540,6 +542,40 @@ function setInputMaxVelocity(lane, value) local(idx)
   setSlider(lane, slider_offset_in_max_velocity, value, 0, 127, 0);
 );
 
+// Move a lane from oldPos to newPos, shifting other lanes to fill holes
+// and preserving the order of shifted lanes.
+function moveLane(oldPos, newPos) local(tmp, direction, currentPos, i, oldIdx, newIdx)
+(
+  (oldPos != newPos && oldPos >= 0 && newPos >= 0 && oldPos < lanes_limit && newPos < lanes_limit) ? (
```

Comment:
Use `num_lanes` instead of `lanes_limit` throughout this function. We only need to rearrange lanes that are visible to the user, which is `num_lanes`. Please also factor out a helper predicate called `activeLane(lane)` that does the bounds checks on this line. Instead of `Pos` suffix for naming locals in this function, please use `Lane`. e.g., oldPos => oldLane.